graphrag/index/flows/generate_text_embeddings.py

```diff
@@ -0,0 +1,153 @@
+    text_units_embeddings = final_text_units.loc[:, ["id", "text"]]
+
+    entities_embeddings = final_entities.loc[:, ["id", "name", "description"]]
+    entities_embeddings["name_description"] = (
+        entities_embeddings["name"] + ":" + entities_embeddings["description"]
+    )
+
+    community_reports_embeddings = final_community_reports.loc[
+        :, ["id", "full_content", "summary", "title"]
+    ]
+
+    embedding_param_map = {
+        document_raw_content_embedding: {
+            "data": documents_embeddings,
+            "column_to_embed": "raw_content",
+            "filename": "create_final_documents_raw_content_embeddings",
+            "base_text_embed": raw_content_text_embed,
+        },
+        relationship_description_embedding: {
+            "data": relationships_embeddings,
+            "column_to_embed": "description",
+            "filename": "create_final_relationships_description_embeddings",
+            "base_text_embed": description_text_embed,
+        },
+        text_unit_text_embedding: {
+            "data": text_units_embeddings,
+            "column_to_embed": "text",
+            "filename": "create_final_text_units_text_embeddings",
+            "base_text_embed": text_text_embed,
+        },
+        entity_name_embedding: {
+            "data": entities_embeddings,
+            "column_to_embed": "name",
+            "filename": "create_final_entities_name_embeddings",
+            "base_text_embed": name_text_embed,
+        },
+        entity_description_embedding: {
+            "data": entities_embeddings,
+            "column_to_embed": "name_description",
+            "filename": "create_final_entities_description_embeddings",
+            "base_text_embed": name_description_text_embed,
+        },
+        community_title_embedding: {
+            "data": community_reports_embeddings,
+            "column_to_embed": "title",
+            "filename": "create_final_community_reports_title_embeddings",
+            "base_text_embed": title_text_embed,
+        },
+        community_summary_embedding: {
+            "data": community_reports_embeddings,
+            "column_to_embed": "summary",
+            "filename": "create_final_community_reports_summary_embeddings",
+            "base_text_embed": summary_text_embed,
+        },
+        community_full_content_embedding: {
+            "data": community_reports_embeddings,
+            "column_to_embed": "full_content",
+            "filename": "create_final_community_reports_full_content_embeddings",
+            "base_text_embed": full_content_text_embed,
+        },
+    }
+
+    log.info("Creating embeddings")
+    for field in embedded_fields:
+        await _run_and_snapshot_embeddings(
+            callbacks=callbacks,
+            cache=cache,
+            storage=storage,
+            **embedding_param_map[field],
+        )
+
+    return pd.DataFrame()
```

Comment:
If this function always returns an empty dataframe why not remove it? looks weird to execute a function to get an empty dataframe... I'm guessing then that _run_and_snapshot_embeddings has a side effect? I'd like to make it clear that this function does not return anything useful by removing the return statement, is that possible?

Reply:
No, according to the return structure workflow is expecting, that's what Nathan explained to me.

Reply:
Andrés has a great point tho. This should be in the verb, not in the core logic. It couples this to datashaper

Reply:
IMHO it makes no sense to have this return here, what is expected is for the verb to return a dataframe, I'd move this pd.DataFrame() directly to the create_verb_output function in the verb to avoid polluting the new code. @natoverse what do you think?

Reply:
@natoverse we talked about this in the past, could you explain to @andresmor-ms please